jax/_src/numpy/polynomial.py

```diff
@@ -57,33 +57,45 @@ def _roots_with_zeros(p: Array, num_leading_zeros: Array | int) -> Array:
   return _where(arange(roots.size) < roots.size - num_leading_zeros, roots, complex(np.nan, np.nan))
 
 
-@implements(np.roots, lax_description="""\
-Unlike the numpy version of this function, the JAX version returns the roots in
-a complex array regardless of the values of the roots. Additionally, the jax
-version of this function adds the ``strip_zeros`` function which must be set to
-False for the function to be compatible with JIT and other JAX transformations.
-With ``strip_zeros=False``, if your coefficients have leading zeros, the
-roots will be padded with NaN values:
-
->>> coeffs = jnp.array([0, 1, 2])
-
-# The default behavior matches numpy and strips leading zeros:
->>> jnp.roots(coeffs)
-Array([-2.+0.j], dtype=complex64)
-
-# With strip_zeros=False, extra roots are set to NaN:
->>> jnp.roots(coeffs, strip_zeros=False)
-Array([-2. +0.j, nan+nanj], dtype=complex64)
-""",
-extra_params="""
-strip_zeros : bool, default=True
-    If set to True, then leading zeros in the coefficients will be stripped, similar
-    to :func:`numpy.roots`. If set to False, leading zeros will not be stripped, and
-    undefined roots will be represented by NaN values in the function output.
-    ``strip_zeros`` must be set to ``False`` for the function to be compatible with
-    :func:`jax.jit` and other JAX transformations.
-""")
 def roots(p: ArrayLike, *, strip_zeros: bool = True) -> Array:
+  r"""Returns the roots of a polynomial with coefficients given in ``p``.
+
+  JAX implementations of :func:`numpy.roots`
+
+  Args:
+    p: Array of polynomial coefficients having rank-1.
+    strip_zeros : bool, default=True
```

Comment:
It looks like this is copied verbatim from the numpy docs. We cannot commit copies of numpy source code into the JAX repository for licensing reasons. All new docs must be written from scratch, without reference to the numpy documentation.

Reply:
Wait, I see this is coming from JAX not from numpy. Nevermind 😁